File: Java/unit2/divideAndConquer/binary_search_nonrecursive.java
package Java.unit2.divideAndConquer;

public class binary_search_nonrecursive {
    static int binsearch(int a[], int key, int low, int high) {
        while (low <= high) {
            int mid = (low + high) / 2;
            if (a[mid] == key) {
                return mid;
            } else if (a[mid] > key) {
                high = mid - 1;

            } else {
                low = mid + 1;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        int a[] = { 11, 25, 39, 44, 59, 67, 79, 98 };
        int ans = binsearch(a, 79, 0, 7);
        System.out.println("key is present at index " + ans);
    }
}

File: C/unit2/divideAndConquer/binary_search_nonrecursive.c
#include <stdio.h>

int binsearch(int *a, int n, int key)
{
    int low = 0, high = n - 1, mid;
    while (low <= high)
    {
        mid = (low + high) / 2;
        if (a[mid] == key)
        {
            return mid;
        }
        else if (a[mid] > key)
        {
            high = mid - 1;
        }
        else
        {
            low = mid + 1;
        }
    }
    return -1;
}

int main()
{
    int a[] = {11, 25, 39, 44, 59, 67, 79, 98};
    int ans = binsearch(a, 8, 79);
    printf("key is present at index %d", ans);
}